src/lib/validate-input.js

```diff
@@ -662,6 +662,68 @@ export function validateScatterElementsParams(input, indices, updates, {axis = 0
   }
 }
 
+export function validateScatterNDParams(input, indices, updates) {
+  // Refer to https://onnx.ai/onnx/operators/onnx__ScatterND.html
+  // ScatterND takes three inputs data tensor of rank r >= 1, indices tensor of rank q >= 1,
+  // and updates tensor of rank q + r - indices.shape[-1] - 1.
```

Comment:
```suggestion
  // and updates tensor of rank equal to input rank + indices rank - indices.shape[-1] - 1.
```

Propose just saying the actual tensor names, as introducing temporary letters of "r" and "q" add no real value to the information and slightly impede comprehensibility with an additional mental lookup (yes, I know those random letters were borrowed from the ONNX doc).